### DISCUSSION TITLE
Holding a child while keyframing it’s parent

### DISCUSSION THREAD

**Discussion by joefiorini** [keiwando/toonsquid#397] [Q&A]
I'm having an issue with key frames. I am not an artist, but I do know my way around vector tools and key frames well enough, but I am not one to sketch an animation frame-by-frame. With that out of the wqy...

I'm trying to make a simple animation of a dancer lifting their leg. I have rigged the person's body parts and have the leg lifting just fine. I also have them rotating their torso back, which may or may not be more realistic, but has led to an interesting problem I'd like to solve. That is: when the torso rotates the supporting leg (the one that isn't kicking) needs to stay in place. Since it's rigged to the torso it insists on rotating with it. How can I keep the leg still while moving the torso?

Edit: currently I'm rotating the support upper leg the same amount as the torso but in the opposite direction. This kind of works but I get a weird snap into position when it hits the keyframe layer. Is this a good approach or is there something better?

**Comment by keiwando** [comment]
For now you have to go with the approach you've mentioned at the end, which is to manually add keyframes to the child layer in order to counter its parent's animation.

This is a case where some sort of IK (inverse kinematics) feature would definitely simplify your workflow, but ToonSquid doesn't have that yet. I'm hoping to add it at some point down the line.
